Common subdirectories: extract-xiso.orig/darwin and extract-xiso/darwin
diff -u extract-xiso.orig/extract-xiso.c extract-xiso/extract-xiso.c
--- extract-xiso.orig/extract-xiso.c	2005-05-21 10:59:45.065431800 +0000
+++ extract-xiso/extract-xiso.c	2005-05-21 10:59:23.935644016 +0000
@@ -1345,6 +1345,11 @@
 	if ( ! err ) {
 		if ( read( in_xiso, dir->filename, dir->filename_length ) != dir->filename_length ) read_err();
 		if ( ! err ) dir->filename[ dir->filename_length ] = 0;
+		if (strstr(dir->filename,"..") || strchr(dir->filename, '/') || strchr(dir->filename, '\\'))
+		  {
+		    printf("Filename contains invalid characters");
+		    exit(1);
+		  }
 	}
 
 	if ( ! err && in_mode == k_generate_avl ) {
Common subdirectories: extract-xiso.orig/libftp-5.0.1.modified.by.in and extract-xiso/libftp-5.0.1.modified.by.in
Common subdirectories: extract-xiso.orig/visual_c++_project and extract-xiso/visual_c++_project
Common subdirectories: extract-xiso.orig/win32 and extract-xiso/win32
